src/genparticles.cxx

```diff
@@ -562,5 +562,56 @@ ROOT::RDF::RNode GenMatching(
     return df3;
 }
 } // end namespace tau
+
+namespace drell_yan {
+
+/**
+ * @brief Extract the flavor of a leptonic Drell-Yan process from the LHE information.
+ * This function checks if exactly two final-state LHE leptons (electron, muon, tau) of
+ * the same flavor are present in the event. If so, the PDG ID of the lepton that the
+ * Z boson or the boson decays into is returned. If no such pair is found, the function
+ * returns -1.
+ * 
+ * Possible values of the returned PDG ID are:
+ * - 11 for a decay into an electron pair,
+ * - 13 for a decay into a muon pair,
+ * - 15 for a decay into a tau pair.
+ * 
+ * @param df input dataframe
+ * @param outputname name of the new column containing the PDG ID of the decay particles
+ * @param lhe_pdgid name of the column containing the PDG ID values of the LHE particles
+ * @param lhe_status name of the column containing the status of the LHE particles
+ * 
+ * @return a new dataframe with the decay flavor column added.
+ */
+ROOT::RDF::RNode DecayFlavor(
+    ROOT::RDF::RNode df,
+    const std::string &outputname,
+    const std::string &lhe_pdgid,
```

Comment:
Did this compile? In the header file the parameter is written with pdg_id, not pdgid.